bft_types/src/lib.rs

```diff
@@ -19,76 +20,114 @@ pub enum Instruction {
+    /// type.
+    ///
+    /// ```
+    ///# use bft_types::Instruction;
+    ///  let c: char = '+';
+    ///
+    ///  let my_instruction = Instruction::from_char(c);
+    /// ```
+    pub fn from_char(c: char) -> Option<Instruction> {
         match c {
-            '<' => Some(Instruction::MoveLeft),
-            '>' => Some(Instruction::MoveRight),
-            '+' => Some(Instruction::Increment),
-            '-' => Some(Instruction::Decrement),
-            '.' => Some(Instruction::Output),
-            ',' => Some(Instruction::Input),
-            '[' => Some(Instruction::ConditionalJumpForward),
-            ']' => Some(Instruction::ConditionalJumpBackward),
+            '<' => Some(Instruction::MoveLeft),
+            '>' => Some(Instruction::MoveRight),
+            '+' => Some(Instruction::Increment),
+            '-' => Some(Instruction::Decrement),
+            '.' => Some(Instruction::Output),
+            ',' => Some(Instruction::Input),
+            '[' => Some(Instruction::ConditionalJumpForward),
+            ']' => Some(Instruction::ConditionalJumpBackward),
             _ => None,
         }
     }
 }
 
-impl Display for Instruction {
+impl Display for Instruction {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let description = match self {
-            Instruction::MoveLeft => "MoveLeft",
-            Instruction::MoveRight => "MoveRight",
-            Instruction::Increment => "Increment",
-            Instruction::Decrement => "Decrement",
-            Instruction::Input => "Input",
-            Instruction::Output => "Output",
-            Instruction::ConditionalJumpForward => "ConditionalJumpForward",
-            Instruction::ConditionalJumpBackward => "ConditionalJumpBackward",
+            Instruction::MoveLeft => "Move tape head left",
+            Instruction::MoveRight => "Move tape head right",
+            Instruction::Increment => "Increment the value in the cell under the head",
+            Instruction::Decrement => "Decrement the value in the cell under the head",
+            Instruction::Input => "Input a byte",
+            Instruction::Output => "Output a byte",
+            Instruction::ConditionalJumpForward => {
+                "Jump forward to the matching ] if the cell is zero"
+            }
+            Instruction::ConditionalJumpBackward => {
+                "Jump backwards to the matching [ if the cell is not zero"
+            }
         };
 
         write!(f, "{}", description)
     }
 }
 
-/// Representation of a Brainfuck instruction, including the instruction type, and the line number and column on which it appears
-#[derive(Debug, PartialEq)]
-pub struct LocalisedInstruction {
+/// A single program [Instruction] with the line and column number it originally appeared on. Line
+/// and column numbers are 1-indexed
+#[derive(Debug, PartialEq, Eq, Clone)]
```

Comment:
If `Instruction` were `Copy` then you could make `LocalisedInstruction` be `Copy` as well.